.github/workflows/hash-backend-cd.yml

```diff
@@ -29,72 +35,56 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
-      - uses: ./.github/actions/docker-ecr-login
+      - name: Docker image build through docker-build-push
+        uses: ./.github/actions/docker-build-push
+        env:
+          HASH_API_IMAGE_NAME: ${{ env.HASH_API_RESOURCE_NAME }}ecr
         with:
+          SHORTNAME: "api"
+          DOCKERFILE_LOCATION: ${{ github.workspace }}/packages/hash/docker/api/prod/Dockerfile
           AWS_ACCESS_KEY_ID: ${{ env.AWS_ACCESS_KEY_ID }}
           AWS_SECRET_ACCESS_KEY: ${{ env.AWS_SECRET_ACCESS_KEY }}
           AWS_REGION: ${{ env.AWS_REGION }}
-
-      # Build images and push to ECR
-      - name: Build and push api
-        uses: docker/build-push-action@v2
-        env:
-          HASH_API_IMAGE_NAME: ${{ env.HASH_API_RESOURCE_NAME }}ecr
-        with:
-          # The Dockerfile is in a leaf-folder without access to what it needs.
-          # Thus the context is within the root, while the dockerfile is deeply nested.
-          context: ${{ github.workspace }}
-          push: true
-          file: ${{ github.workspace }}/packages/hash/docker/api/prod/Dockerfile
-          tags: ${{ env.AWS_ECR_URL }}/${{ env.HASH_API_IMAGE_NAME }}:latest , ${{ env.AWS_ECR_URL }}/${{ env.HASH_API_IMAGE_NAME }}:run-${{ env.GH_RUN_ID }}
+          AWS_ECR_URL: ${{ env.AWS_ECR_URL }}
```

Comment:
Not something introduced by this PR, but looking at it makes me think we should scope this better at some point to allow for usage of other ECRs in the repo, e.g. consistent `HASH_` prefixes.

Reply:
I think all ECR repositories we make within our AWS tenant would have that same ECR URL as the URL contains the AWS account id. So having other ECRs would also mean other HASH-wide accounts in AWS, which I'm not sure we're going to do down the line

Reply:
My bad, sorry – I was thinking `AWS_ECR_URL` contains the repository name, but that's in `X_IMAGE_NAME`